Makefile

```diff
@@ -1,22 +1,97 @@
-PACKAGES := $(shell glide novendor)
+PKGS := $(shell go list ./... | grep -v go.uber.org/sally/vendor)
+SRCS := $(wildcard *.go)
 
-.PHONY: install
-install:
-	glide --version || go get github.com/Masterminds/glide
+.PHONY: all
+all: test
+
+.PHONY: vendor-update
+vendor-update:
+	rm -rf vendor
+	go get -d -v -t -u -f ./...
+	go get -v github.com/Masterminds/glide
+	glide create
+	glide update
+
+.PHONY: vendor-install
+vendor-install:
+	go get -v github.com/Masterminds/glide
 	glide install
 
+.PHONY: build
+build:
+	go build $(PKGS)
+
+.PHONY: install
+install:
+	go install $(PKGS)
 
 .PHONY: lint
 lint:
-	go vet $(PACKAGES)
-	$(foreach pkg, $(PACKAGES), golint $(pkg) &&) echo "success"
+	go get -v github.com/golang/lint/golint
+	for file in $$(find . -name '*.go' | grep -v '^\./vendor'); do \
+		golint $$file; \
+		if [ -n "$$(golint $$file)" ]; then \
+			exit 1; \
+		fi; \
+	done
+
+.PHONY: vet
+vet:
+	go vet $(PKGS)
+
+.PHONY: errcheck
+errcheck:
+	go get -v github.com/kisielk/errcheck
+	errcheck $(PKGS)
 
+.PHONY: staticcheck
+staticcheck:
```

Comment:
I think it's a lot nicer to have them as separate targets, especially if you are debugging one that is failing.